### PR TITLE
[SR SQL Planner Summer Camp] Support truncate table in new parser

### PR DESCRIPTION
**Pull request by xlwh**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7938

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 31 / 43 (72.09%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/TruncateTableAnalyzer.java|13|22|59.09%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|4|5|80.00%|
|:large_blue_circle:|com/starrocks/analysis/TruncateTableStmt.java|6|7|85.71%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|6|100.00%|

**Comment by xlwh** [comment]
run starrocks_fe_unittest

**Comment by xlwh** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 31 / 43 (72.09%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/TruncateTableAnalyzer.java|13|22|59.09%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|4|5|80.00%|
|:large_blue_circle:|com/starrocks/analysis/TruncateTableStmt.java|6|7|85.71%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|6|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/TruncateTableAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import com.starrocks.cluster.ClusterNamespace;
+import com.starrocks.common.AnalysisException;
+import com.starrocks.common.ErrorCode;
+import com.starrocks.common.ErrorReport;
+import com.starrocks.qe.ConnectContext;
+
+public class TruncateTableAnalyzer {
+
+    public static void analyze(TruncateTableStmt statement, ConnectContext context) {
+        String dbName = statement.getDbName();
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/TruncateTableAnalyzer.java:18]: MetaUtils.normalizationTableName(statement.getTblRef().getName()) is enough？

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 21 / 32 (65.62%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[280]|
|:large_blue_circle:|com/starrocks/analysis/TruncateTableStmt.java|2|7|28.57%|[40, 44, 45, 48, 52]|
|:large_blue_circle:|com/starrocks/sql/analyzer/TruncateTableAnalyzer.java|7|11|63.64%|[11, 16, 22, 26]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|4|5|80.00%|[329]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|6|100.00%|[]|

**Comment by xlwh** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 21 / 32 (65.62%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[285]|
|:large_blue_circle:|com/starrocks/analysis/TruncateTableStmt.java|2|7|28.57%|[40, 44, 45, 48, 52]|
|:large_blue_circle:|com/starrocks/sql/analyzer/TruncateTableAnalyzer.java|7|11|63.64%|[11, 16, 22, 26]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|4|5|80.00%|[330]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|6|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 23 / 31 (74.19%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[285]|
|:large_blue_circle:|com/starrocks/sql/analyzer/TruncateTableAnalyzer.java|7|11|63.64%|[11, 16, 22, 26]|
|:large_blue_circle:|com/starrocks/analysis/TruncateTableStmt.java|4|6|66.67%|[44, 45]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|4|5|80.00%|[330]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|6|100.00%|[]|

**Comment by xlwh** [comment]
run starrocks_admit_test

**Pull request merged**